src/superphot_plus/plotting/sampling_results.py

```diff
@@ -0,0 +1,455 @@
+        bin_centers = (bin_edges[1:] + bin_edges[:-1]) / 2
+
+        for a in allowed_types:
+            # a = labels_to_classes[at]
+            features_1_t = features_1_gauss[labels_gaussian == a]
+
+            if log_scale:
+                feature_hist, bin_edges = np.histogram(features_1_t, bins=bin_edges)
+                current_area = np.sum(bin_width * feature_hist)
+                feature_hist = (1.0 / current_area) * feature_hist
+            else:
+                feature_hist, bin_edges = np.histogram(features_1_t, bins=bin_edges, density=True)
+
+            feature_hist[np.abs(feature_hist) > 1e5] = 0
+            (l,) = axes[ax_num].step(bin_centers, feature_hist, where="mid", label=a)
+            leg_lines.append(l)
+
+        ax = axes[ax_num]
+        (l,) = ax.step(bin_centers, feature_hist_all, where="mid", c="k", label="Combined", linewidth=2)
+        leg_lines.append(l)
+
+        if log_scale:
+            bins_fine = np.linspace(np.log10(bin_centers[0]), np.log10(bin_centers[-1]), num=100)
+            prior_dist = gaussian(bins_fine, 1.0, prior_means[i], prior_stddevs[i])
+            bins_fine = 10**bins_fine
+        else:
+            bins_fine = np.linspace(bin_centers[0], bin_centers[-1], num=100)
+            prior_dist = gaussian(bins_fine, 1.0, prior_means[i], prior_stddevs[i])
+
+        (l,) = ax.plot(bins_fine, prior_dist, linestyle="dashed", label="Prior", linewidth=2, c="magenta")
+        leg_lines.append(l)
+        ax.set_xlabel(param_1)
+        ax.set_yticklabels([])
+        ax.set_yticks([])
+
+        ratio = 1.25
+        x_left, x_right = ax.get_xlim()
+        y_low, y_high = ax.get_ylim()
+
+        if log_scale:
+            ax.set_aspect(abs(np.log10(x_right / x_left) / (y_low - y_high)) * ratio)
+        else:
+            ax.set_aspect(abs((x_right - x_left) / (y_low - y_high)) * ratio)
+
+        ax_num += 1
+
+    fig.legend(leg_lines, [*allowed_types, "Combined", "Prior"], loc="lower center", ncol=4)
+    plt.locator_params(axis="x", nbins=3)
+
+    plt.savefig(os.path.join(save_dir, "all_1d_hists.pdf"), bbox_inches="tight")
+    plt.close()
+
+
+def plot_combined_posterior_space(names, labels, fits_dir, save_dir):
+    """
+    Plot 2D scatterplots for each pair
+    of fit parameters, to identify clustering
+    among different subclasses.
+
+    TODO: modify to plot different points for each type.
+
+    Parameters
+    ----------
+    fits_fn : str
+        File path for fit posteriors.
+    save_dir : str
+        Where to save figure
+    """
+    print(save_dir)
+    os.makedirs(os.path.join(save_dir, "combined_2d_posteriors"), exist_ok=True)
+    labels_to_class, classes_to_labels = SnClass.get_type_maps()
+    # pt_colors = ["r", "c", "k", "m", "g"] # keep for TODO
+
+    features, labels = oversample_using_posteriors(names, labels, OVERSAMPLE_SIZE, fits_dir)
+
+    print(names)
+    params, save_labels = param_labels(
+        [
+            "g",
```

Comment:
I'm just going to comment on all the lines like this, so I look more productive =]